src/lib/plot/contour.typ

```diff
@@ -5,11 +5,17 @@
 // Find contours of a 2D array by using marching squares algorithm
 //
 // - data (array): 2D float array (rows => columns)
-// - offset (float): Value (offset) a cell must be greater or equal to to count as true
+// - offset (float): Z value (offset) of a cell compare with `op` to, to count as true
+// - op (auto,string,funciton): Z value comparison oparator:
+//     / `">", ">=", "<", "<=", "!=", "=="`: Use the passed operator to compare z.
+//     / `auto`: Use ">=" for positive z values, "<=" for negative z values.
+//     / `<function>`: If set to a function, that function gets called
+//       with two arguments, the z value `z1` to compare against and
+//       the z value `z2` of the data and must return a boolean: `(z1, z2) => boolean`.
 // - interpolate (bool): Enable cell interpolation for smoother lines
 // - contour-limit (int): Contour limit after which the algorithm panics
 // -> array: Array of contour point arrays
-#let find-contours(data, offset, interpolate: true, contour-limit: 50) = {
+#let find-contours(data, offset, op: auto, interpolate: true, contour-limit: 50) = {
   assert(data != none)
   assert(type(data) == array)
   assert(type(offset) in (int, float))
```

Comment:
I'm not sure if this is relevant to the pr but error messages should be added here